src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -484,15 +474,21 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
     const options = this.#options;
     const { block, blockLogs, transactions } = blockInfo;
 
-    // emit the block once everything has been fully saved to the database
     transactions.forEach(transaction => {
       transaction.finalize("confirmed", transaction.execException);
     });
 
-    if (this.#instamine && options.miner.legacyInstamine) {
-      // in legacy instamine mode we must delay the broadcast of new blocks
+    if (this.#instamine && options.miner.instamine === "greedy") {
+      // in greedy instamine mode we must delay the broadcast of new blocks
       await new Promise(resolve => {
-        process.nextTick(async () => {
+        // we delay emitting blocks and blockLogs because we need to allow for:
+        // ```
+        //  provider.request({"method": "eth_sendTransaction"...)
+        //  provider.once("message")
+        // ```
+        // If we don't have this delay here the messages will be sent before
+        // the call has a chance to listen to the event.
+        setImmediate(async () => {
```

Comment:
It turns out this was broken in `greedy` mode!

Reply:
what is broken?

Reply:
The delay wasn't working. Our existing tests caught it when I switch the default.

```
await provider.send("eth_sendTransaction", ...);
await provider.once("message"); // <- this wasn't work, but it should.
```